globus_action_provider_tools/data_types.py

```diff
@@ -84,31 +86,93 @@ class ActionProviderDescription:
     event_types: Optional[List[EventType]] = None
 
 
-@dataclass
-class ActionRequest:
-    request_id: str
-    body: Dict[str, Any]
-    label: Optional[str] = None
-    monitor_by: Optional[List[str]] = None
-    manage_by: Optional[List[str]] = None
-    allowed_clients: Optional[List[str]] = None
-    deadline: Optional[str] = None
-    release_after: Optional[str] = None
-
-
-@dataclass
-class ActionStatus:
-    status: ActionStatusValue
-    creator_id: str
-    action_id: str = field(default_factory=shortish_id)
-    start_time: str = field(default_factory=now_isoformat)
-    label: Optional[str] = None
-    monitor_by: Optional[Iterable[str]] = None
-    manage_by: Optional[Iterable[str]] = None
-    completion_time: Optional[str] = None
-    release_after: Optional[str] = None
-    display_status: Optional[str] = None
-    details: Optional[Dict[str, Any]] = None
+class ActionRequest(BaseModel):
```

Comment:
If this is no longer a dataclass, do you know if our code in our actual providers which I believe uses the dataclass-based method of creating a dict for storage in dynamo will still work?

pydantic provides a dataclass implementation as well which I think blends fine with their BaseModel. Perhaps decorating with the pydantic dataclass implementation gets us both behaviors we want.

Reply:
This is a good question, I just did a search on `globus-automate` to remind myself how we're dumping the `ActionStatus` objects and there's two things going on -> we `asdict()` it and dump it using the `ActionProviderJsonEncoder`. If we're using the custom encoder, we should be good on that since it knows how to handle pydantic types. The `asdict` part I'm not so sure about but we can get rid of that since the custom encoder also handles dumping dataclasses.

Reply:
I just tried this out, if we try to run asdict() on an instance of a `pydantic` `ActionStatus`, we get `TypeError: asdict() should be called on dataclass instances`. We're good if we just remove the call to `asdict` in the `json.dump`.